addons/im_livechat/controllers/cors/main.py

```diff
@@ -15,7 +15,11 @@ def cors_visitor_leave_session(self, guest_token, uuid):
     def cors_get_session(
         self, channel_id, anonymous_name, previous_operator_id=None, chatbot_script_id=None, persisted=True, **kwargs
     ):
-        downgrade_to_public_user()
+        guest_token = kwargs.get("guest_token")
+        if guest_token:
+            force_guest_env(guest_token)
+        else:
+            downgrade_to_public_user()
```

Comment:
I would not raise on invalid token here, we should be able to create a new session no matter what (obsolete token or whatever).